server/src/routes/index.ts

```diff
@@ -10,30 +10,382 @@ import passport from 'passport';
 import cors from 'cors';
 import cookieParser from 'cookie-parser';
 import { frontendOrigin } from '../configs';
+import { definition } from './swagger';
+
+import swaggerUI from 'swagger-ui-express';
+import swaggerJsdoc from 'swagger-jsdoc';
+
+const options: object = {
+  definition,
+  apis: ['./src/routes/*.ts'],
+};
+
+const openapiSpecification: object = swaggerJsdoc(options);
 
 export const router: Router = Router();
 
 router.use(cookieParser());
 
+router.use('/api-docs', swaggerUI.serve);
```

Comment:
I don't see why this line is here, is it necessary?

Reply:
https://www.npmjs.com/package/swagger-ui-express, according to this documentation

Reply:
`router.use` mounts a middleware function to all requests to a specific path (https://expressjs.com/en/4x/api.html#app.use, this is for `app.use` but `router.use` should work the same way), in this case `swaggerUI.serve` to all requests to `/api-docs`.

In the following line you're also defining `swaggerUI.serve` to be used for GET requests to `/api-docs`, which is redundant. So unless `swaggerUI.serve` is needed for other requests which haven't been explicitly defined here (yet?), then I don't think this line is needed. Otherwise the following line should not include `swaggerUI.serve`.

Reply:
Ok now I understand your point. The router.get function incorrectly contained the swaggerUI.serve function.

Swagger doesn't work without stating router.use for the middleware.